internal/pkg/update/github_test.go

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package update
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLatestGithub(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+
+	t.Parallel()
+
+	c := newGitHub(gitHubTokenFromEnv())
+
+	for source, expected := range map[string]*LatestInfo{
+		// https://github.com/pullmoll/musl-fts/releases has only tags.
+		"https://github.com/pullmoll/musl-fts/archive/refs/tags/v1.2.6.tar.gz": {
+			HasUpdate: true,
+			BaseURL:   "https://github.com/pullmoll/musl-fts/releases/",
+			LatestURL: "https://github.com/pullmoll/musl-fts/archive/refs/tags/v1.2.7.tar.gz",
```

Comment:
This the latest _today_, but what happens when a new release comes out? IIUC, these tests will fail eventually?

Reply:
Yes, they will fail and should be updated.

I tried to peek repositories that do not often change instead of using mocks or something – I prefer "real" tests.